staging/src/k8s.io/client-go/pkg/apis/clientauthentication/v1beta1/types_test.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"reflect"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	clientcmdv1 "k8s.io/client-go/tools/clientcmd/api/v1"
+)
+
+// TestCluster ensures that Cluster stays in sync with clientcmd.Cluster.
+//
+// We want Cluster to offer the same knobs as clientcmdv1.Cluster to allow someone to connect to
+// the kubernetes API. This test should fail if a new field is added to one of the structs without
+// updating the other.
+func TestCluster(t *testing.T) {
```

Comment:
This test looks good for catching gaps in both directions.

would this make more sense in staging/src/k8s.io/client-go/tools/auth/exec which already has visibility to both the `k8s.io/client-go/tools/clientcmd/api` and `k8s.io/client-go/pkg/apis/clientauthentication` API packages? it seems like it would pair well with the test that ensures we can construct rest.Config objects with fidelity from the envvar.

Reply:
Sure, sounds reasonable to me. I was nervous to move it to `.../tools/...` because I didn't know if there was a convention around testing API structs as close to their declaration as possible, but I totally agree with you that `k8s.io/client-go/tools/clientcmd/api` seems like the natural place for this since the dependency graph works out nicely.

Reply:
502dd3fcf0d